### PR TITLE
Add curl cloud builder

### PR DESCRIPTION
**Pull request by palmerj3**
Add curl cloud builder. Based off the wget cloud builder.

### CODE REVIEW

**Comment by rovolution** [comment]
👍

File: curl/cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+- name: gcr.io/cloud-builders/docker
+  args: ['build', '--tag=gcr.io/$PROJECT_ID/curl', '.']
+
+# Print version information.
+- name: gcr.io/$PROJECT_ID/curl
+  args: ['-V']
+
+# GET data from a server, specifying an Authorization header.
+- name: gcr.io/$PROJECT_ID/curl
+  args: ['-O=file.out', '--header="Authorization: Bearer foobar"', 'https://www.example.com']
```
[the remark this answers is not in the document]
Comment (palmerj3) [reply curl/cloudbuild.yaml]: Sure!

**Comment by rovolution** [comment]
@palmerj3 reminder to update the main `README` and add it to the list - https://github.com/GoogleCloudPlatform/cloud-builders/blob/master/README.md

**Comment by palmerj3** [comment]
ping @ImJasonH for one last review

**Comment by rovolution** [comment]
any progress on this getting merged? Thanks!

**Comment by ImJasonH** [comment]
I checked this out locally to try to build it, and got:

```
Step #2: Already have image: gcr.io/test-argo/curl
Step #2: curl: option --header="Authorization: Bearer foobar": is unknown
Step #2: curl: try 'curl --help' or 'curl --manual' for more information
```

I got it to work by changing the last steps to:

```
# GET data from a server, specifying an Authorization header.
- name: gcr.io/$PROJECT_ID/curl
  args: ['-o', 'file.out', "--header", "Authorization: Bearer foobar'", 'https://www.example.com']

# POST information to a server, specifying a Content-type header.
- name: gcr.io/$PROJECT_ID/curl
  args: ['--header', 'Content-type: application/json"', '--data', '"{\"buildID\": \"$BUILD_ID\"}"', 'https://www.example.com']
```

**Comment by ImJasonH** [comment]
@palmerj3 I've pushed a commit to fix your cloudbuild.yaml, let me know if you'd like to make any other changes, or if you don't need this image after all.

**Comment by palmerj3** [comment]
@ImJasonH thank you! It's not mission critical but would definitely help engineers if it were available :)

I don't have any further changes to request.

Thanks again

**Comment by palmerj3** [comment]
LGTM

**Pull request merged**